app/notification_test.go

```diff
@@ -68,6 +68,53 @@ func TestSendNotifications(t *testing.T) {
 	mentions, err = th.App.SendNotifications(post1, th.BasicTeam, th.BasicChannel, th.BasicUser, nil, true)
 	require.NoError(t, err)
 	require.Empty(t, mentions)
+
+	t.Run("replies to post created by OAuth bot should not notify user", func(t *testing.T) {
+		th := Setup(t).InitBasic()
+		defer th.TearDown()
+		testUserNotNotified := func(t *testing.T, user *model.User) {
+			rootPost := &model.Post{
+				UserId:    user.Id,
+				ChannelId: th.BasicChannel.Id,
+				Message:   "a message",
+				Props:     model.StringInterface{"from_webhook": "true", "override_username": "a bot"},
+			}
+
+			rootPost, appErr := th.App.CreatePostMissingChannel(rootPost, false)
+			require.Nil(t, appErr)
+
+			childPost := &model.Post{
+				UserId:    th.BasicUser2.Id,
+				ChannelId: th.BasicChannel.Id,
+				RootId:    rootPost.Id,
+				Message:   "a reply",
+			}
+			childPost, appErr = th.App.CreatePostMissingChannel(childPost, false)
+			require.Nil(t, appErr)
+
+			postList := model.PostList{
+				Order: []string{rootPost.Id, childPost.Id},
+				Posts: map[string]*model.Post{rootPost.Id: rootPost, childPost.Id: childPost},
+			}
+			mentions, err := th.App.SendNotifications(childPost, th.BasicTeam, th.BasicChannel, th.BasicUser2, &postList, true)
```

Comment:
This line is causing the linter to fail due to [shadowing](https://yourbasic.org/golang/gotcha-shadowing-variables/) the `err` variable.
What that means is that we are declaring it again when we already had one defined before, in this case at line 32.
If it helps you can run the `make check-style` locally to easily check for these errors.

Reply:
Thanks :slightly_smiling_face: